client/components/infinite-scroll/index.js

```diff
@@ -20,12 +20,17 @@ export default class InfiniteScroll extends Component {
 	);
 
 	componentDidMount() {
+		this.initialMount = true;
 		window.addEventListener( 'scroll', this.checkScrollPositionHandler );
-		this.checkScrollPosition( false );
+		defer( () => this.checkScrollPosition( false ) );
```

Comment:
The component could unmount before the scroll position check runs. We should either stash the timer handle and cancel it in componentWillUnmount

Reply:
`defer` returns a timer ID that can be passed to `clearTimeout`.

Reply:
That's a very good point, and it applies to both components. Thanks! I'll make that change.